### ISSUE TITLE
check_http asks for HTTP/1.1, but doesn't understand chunked [sf#2786808]

### ISSUE THREAD

**Issue by monitoring-user** [monitoring-plugins/monitoring-plugins#1122]
Submitted by wferi on 2009-05-04 21:19:43

Fixing 2638765 uncovered this bug (actually, most HTTP servers respect Host: headers in HTTP/1.0 requests, even though it isn't part of the spec).
Anyway: if a client sends a HTTP/1.1 request, it must be able to handle the incoming HTTP/1.1 reply. Which includes coping with a chunked one.
This exact same command worked with check_http version 1.4.5 (and still does so with -H replaced by -I, but that isn't the point).
```
$ /usr/lib/nagios/plugins/check_http -v -H idp.niif.hu -u /idp/profile/Status --ssl -r "^ok$"
GET /idp/profile/Status HTTP/1.1
Host: idp.niif.hu:443
User-Agent: check_http/v1991 (nagios-plugins 1.4.12)
Connection: close

https://idp.niif.hu:443/idp/profile/Status is 141 characters
STATUS: HTTP/1.1 200 OK
***\* HEADER ****
Date: Mon, 04 May 2009 19:12:13 GMT
Connection: close
Transfer-Encoding: chunked
Content-Type: text/plain
***\* CONTENT ****
2
ok
0

HTTP CRITICAL - pattern not found|time=0.067933s;;;0.000000 size=141B;;;0
```

**Comment by alexs77** [comment]
It's 2015 and the problem still exists…
As I did not find this issue here, I reported it in [nagios-plugins: check_http - Chunked Transfer Encoding leaks TCP header text into checked text, causing content checks to fail #76](https://github.com/nagios-plugins/nagios-plugins/issues/76)… :(

**Comment by koenwtje** [comment]
This has already been fixed in the 2.1 branch: https://github.com/nagios-plugins/nagios-plugins/commit/890d124e3ad908b0e553eaa6ef83b3b80c5a2dd9,  https://github.com/nagios-plugins/nagios-plugins/commit/f5cd931d01965991850d1e621d8f986ea1fd7f90

**Comment by glensc** [comment]
i do not think this applies to monitoring-plugins, nagios-plugins added this in [2.1.0](https://nagios-plugins.org/nagios-plugins-2-1-0-released/), but this project never got it according to my diffs:
https://gist.github.com/glensc/f2f0dfa36d2e369a6493

which is good as the implementation i see is very broken, internet is full of the bugreports and np project does nothing to handle them:
  - https://sourceforge.net/p/nagiosplug/bugs/446/
  - https://support.nagios.com/forum/viewtopic.php?f=35&t=34548
  - https://github.com/nagios-plugins/nagios-plugins/issues/103
  - https://github.com/nagios-plugins/nagios-plugins/issues/98
  - https://github.com/nagios-plugins/nagios-plugins/issues/104
  - https://github.com/nagios-plugins/nagios-plugins/issues/121
  - https://github.com/nagios-plugins/nagios-plugins/issues/76

only one i see related to this is unmerged PR to add chunked encoding to this mp project: #1286
